### PR TITLE
NMS-8624: Make enlinkd location aware

### PR DESCRIPTION
**Pull request by rssntn67**
NMS-8624: Make enlinkd location aware

Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS904-3

### CODE REVIEW

File: opennms-services/src/main/java/org/opennms/netmgt/enlinkd/NodeDiscoveryBridge.java [hunk 1]
```diff
@@ -142,79 +141,74 @@ protected void runCollection() {
...
+            final Dot1dBaseTracker dot1dbase = new Dot1dBaseTracker();
+            
+            try {
+                m_linkd.getLocationAwareSnmpClient().walk(getPeer(), dot1dbase).
+                withDescription("dot1dbase").
+                atLocation(getLocation()).
+                execute().
+                get();
+           } catch (ExecutionException e) {
+                    // pass
```
Comment (j-white) [inline opennms-services/src/main/java/org/opennms/netmgt/enlinkd/NodeDiscoveryBridge.java]: We should log the exception, and treat it similar to the timeout behavior we had previously.

**Comment by mvrueden** [comment]
Can we actually make it one commit, or distinguish the commit messages.

**Comment by rssntn67** [comment]
Safely Merged after bamboo get green

File: opennms-services/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdSnmpIT.java [hunk 2]
```diff
@@ -173,17 +177,13 @@ public void testCdpGlobalGroupCollection() throws Exception {
...
         try {
-            walker.waitFor();
-            if (walker.timedOut()) {
-                LOG.info("run:Aborting Cdp Linkd node scan : Agent timed out while scanning the {} table", trackerName);
-            }  else if (walker.failed()) {
-                LOG.info("run:Aborting Cdp Linkd node scan : Agent failed while scanning the {} table: {}", trackerName,walker.getErrorMessage());
-            }
+            m_client.walk(config,cdpGlobalGroup)
+            .withDescription(trackerName)
+            .atLocation(null)
```
Comment (j-white) [inline opennms-services/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdSnmpIT.java]: Location should not be null here.

File: opennms-services/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdSnmpIT.java [hunk 3]
```diff
@@ -205,17 +205,13 @@ public void testCdpGlobalGroupCollectionWithGlobalIdFormat() throws Exception {
...
```
Comment (j-white) [inline opennms-services/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdSnmpIT.java]: Location should not be null here.

File: opennms-services/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdSnmpIT.java (hunk 2 above)
Comment (j-white) [inline opennms-services/src/test/java/org/opennms/netmgt/enlinkd/EnLinkdSnmpIT.java]: Nevermind, I just realized this is a test.

**Pull request merged**